### ISSUE TITLE
Filter between device architecture

### ISSUE THREAD

**Issue by ajstrongdev** [pacstall/pacstall#983]
### Is your feature related to a problem? Please describe.

Rhino Linux provides builds for both x86_64 and ARM64 CPU architectures. 

The ability to filter between these on both the website and while utilising `pacstall -I` or `pacstall -S` would go a long way in improving having to hunt down packages and then ensure it will work on your device before attempting to install.

### A clear and concise description of what you want to happen.

On the website, a filters section would be nice, and could be expanded to include `deb`, `appimage` and `bin` filters. The user would select their architecture, if it supports both then it will still show up, and then they would be able to see what packages they can install based on architecture.

Within Pacstall itself, a new flag could be added to look for name and then architecture such as `pacstall -F discord arm64`

This will search for packages with descriptions that contain `discord` and work on `arm64`

**Comment by Elsie19** [comment]
It's possible on the site, but not on pacstall without a major change in the way searching works, and I don't want to change it to source every single package it finds for information.

**Comment by oklopfer** [comment]
@Elsie19 this would actually be incredibly easy to do without changing how pacstall does things too much. 

Instead of directly making the function in pacstall, it would probably be a combination of rpk+pacstall, since pacstall can't parse just `pacstall -I discord`, it's `pacstall -I discord-deb` or `rpk install {-d} discord`. Here's what I imagine:

1. In Pacstall, create a new flag `--arch` so that `pacstall -I ${package} --arch ${INPUTARCH}`. Then, instead of doing what we did in ${FARCH}, where we defined an alternative if ${CARCH} doesn't match, we can instead provide a redefinition of ${CARCH} where:
```bash
if [[ -n ${INPUTARCH} ]]; then
  CARCH=${INPUTARCH}
else
  CARCH=(whatever we currently do here)
fi
```
This would work only with the defined package name though, or with pacstall -S - which is true for the command without `--arch`, anyway. 

2. Create a similar flag `rpk install ${package} --arch ${ARCH}` which can send the flag to pacstall, and then AJ can run `rpk install discord --arch amd64`
